x-pack/plugin/ent-search/src/main/java/org/elasticsearch/xpack/application/connector/Connector.java

```diff
@@ -0,0 +1,443 @@
+            syncInfo.toXContent(builder, params);
+            builder.field(NAME_FIELD.getPreferredName(), name);
+            builder.field(PIPELINE_FIELD.getPreferredName(), pipeline);
+            builder.field(SCHEDULING_FIELD.getPreferredName(), scheduling);
+            builder.field(SERVICE_TYPE_FIELD.getPreferredName(), serviceType);
+            builder.field(SYNC_CURSOR_FIELD.getPreferredName(), syncCursor);
+            builder.field(STATUS_FIELD.getPreferredName(), status.toString());
+            builder.field(SYNC_NOW_FIELD.getPreferredName(), syncNow);
+        }
+        builder.endObject();
+        return builder;
+    }
+
+    @Override
+    public void writeTo(StreamOutput out) throws IOException {
+        out.writeString(connectorId);
+        out.writeOptionalString(apiKeyId);
+        out.writeMap(configuration, StreamOutput::writeString, StreamOutput::writeGenericValue);
+        out.writeOptionalWriteable(customScheduling);
+        out.writeOptionalString(description);
+        out.writeOptionalString(error);
+        out.writeOptionalWriteable(features);
+        out.writeOptionalCollection(filtering);
+        out.writeOptionalString(indexName);
+        out.writeOptionalBoolean(isNative);
+        out.writeOptionalString(language);
+        out.writeOptionalWriteable(syncInfo);
+        out.writeOptionalString(name);
+        out.writeOptionalWriteable(pipeline);
+        out.writeOptionalWriteable(scheduling);
+        out.writeOptionalString(serviceType);
+        out.writeEnum(status);
+        out.writeGenericValue(syncCursor);
+        out.writeOptionalBoolean(syncNow);
+    }
+
+    public String getConnectorId() {
+        return connectorId;
+    }
+
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) return true;
+        if (o == null || getClass() != o.getClass()) return false;
+        Connector connector = (Connector) o;
+        return Objects.equals(connectorId, connector.connectorId)
+            && Objects.equals(apiKeyId, connector.apiKeyId)
+            && Objects.equals(configuration, connector.configuration)
+            && Objects.equals(customScheduling, connector.customScheduling)
+            && Objects.equals(description, connector.description)
+            && Objects.equals(error, connector.error)
+            && Objects.equals(features, connector.features)
+            && Objects.equals(filtering, connector.filtering)
+            && Objects.equals(indexName, connector.indexName)
+            && Objects.equals(isNative, connector.isNative)
+            && Objects.equals(language, connector.language)
+            && Objects.equals(syncInfo, connector.syncInfo)
+            && Objects.equals(name, connector.name)
+            && Objects.equals(pipeline, connector.pipeline)
+            && Objects.equals(scheduling, connector.scheduling)
+            && Objects.equals(serviceType, connector.serviceType)
+            && status == connector.status
```

Comment:
question: just curious, why is this a different equality check? It's not the only string in here so I would guess that's not it.

Reply:
This is autogenerated by Intellij. The status is represented internally as enum. I asked chatgpt about it, here is the response (I'm not even pretending that I'm such a Java expert 😅  )

> When comparing enum values in Java, it's best to use the == operator rather than Object.equals. This is because enums in Java are inherently singletons, meaning that there's only one instance of each enum constant. Due to this singleton nature, using == is safe and more performant for comparing enum values, as it compares the object references directly.